File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--programmatically-list-all-worksheets-in-a-workbook_1.cs
        private void ListSheets()
        {
            int index = 0;

            Microsoft.Office.Tools.Excel.NamedRange NamedRange1 =
                Globals.Sheet1.Controls.AddNamedRange(
                Globals.Sheet1.Range["A1"], "NamedRange1");

            foreach (Excel.Worksheet displayWorksheet in Globals.ThisWorkbook.Worksheets)
            {
                NamedRange1.Offset[index, 0].Value2 = displayWorksheet.Name;
                index++;
            }
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--programmatically-list-all-worksheets-in-a-workbook_1.vb
    Private Sub ListSheets()
        Dim index As Integer = 0

        Dim NamedRange1 As Microsoft.Office.Tools.Excel.NamedRange = _
            Globals.Sheet1.Controls.AddNamedRange( _
            Globals.Sheet1.Range("A1"), "NamedRange1")

        For Each displayWorksheet As Excel.Worksheet In Globals.ThisWorkbook.Worksheets
            NamedRange1.Offset(index, 0).Value2 = displayWorksheet.Name
            index += 1
        Next displayWorksheet
    End Sub